File: WSConsumer/src/ifbarbacena/taroman/ws/ex2/GITConsumer.java
package ifbarbacena.taroman.ws.ex2;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.URL;
import java.net.URLConnection;
import java.util.Iterator;

import javax.swing.JOptionPane;

import org.json.simple.JSONArray;
import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;
import org.json.simple.parser.ParseException;

public class GITConsumer {
	public static void main(String[] args) {

		String semiURL = "https://api.github.com/users/";

		String userName = lerString("Digite o nome de usuário: \n");
		if(userName == null) return;

		do {
			try {
				String source = doRequest(semiURL+userName);
				//System.out.println(source.toString());
				JSONObject jso = getJSONObject(source);
				
				StringBuilder result = new StringBuilder();
				
				result.append("Usuário da consulta: ").append(jso.get("name")).append("\n\n");
				result.append(jso.get("followers")).append(" seguidores").append("\n\n");
				
				source = doRequest((String)jso.get("followers_url"));
				//System.out.println(source.toString());
				JSONArray seguidores = getJSONArray(source);
				Iterator<JSONObject> iteSeg = seguidores.iterator();
				
				JSONObject jsoAux;
				JSONArray repositorios;
				while(iteSeg.hasNext()) {
					jsoAux = iteSeg.next();
					//System.out.println(jsoAux.get("url"));
					
					source = doRequest((String) jsoAux.get("url"));
					jsoAux = getJSONObject(source);
					
					result.append(jsoAux.get("name")).append("\n");
					
					source = doRequest((String) jsoAux.get("repos_url"));
					repositorios = getJSONArray(source);
					Iterator<JSONObject> iteRep = repositorios.iterator();
					
					while(iteRep.hasNext()) {
						jsoAux = iteRep.next();
						result.append("\t").append("name");
					}
					
				}
				
				System.out.println(result.toString());
			} catch (IOException | ParseException e) {
				e.printStackTrace();
				JOptionPane.showMessageDialog(null, "Não encontrou usuário ou erro na API");
			}	

			userName = lerString("Digite o nome de usuário: \n");
		}while(userName != null);


	}

	public static String lerString(String mensagem) {
		return JOptionPane.showInputDialog(mensagem);
	}
	
	public static String doRequest(String strUrl) throws IOException {
		URL url = new URL(strUrl);
		URLConnection con = url.openConnection();

		BufferedReader input = new BufferedReader(new InputStreamReader(con.getInputStream(), "utf-8"));
		String line;
		StringBuilder source = new StringBuilder();
		while((line = input.readLine()) != null)
			source.append(line);
		input.close();
		
		return source.toString();
	}
	
	public static JSONObject getJSONObject(String strJSON) throws ParseException {
		JSONParser parse = new JSONParser();
		return (JSONObject) parse.parse(strJSON);
	}
	
	public static JSONArray getJSONArray(String strJSON) throws ParseException {
		JSONParser parse = new JSONParser();
		return (JSONArray) parse.parse(strJSON);
	}
}

File: WSConsumer/GITConsumer - Python/GITConsumer.py
#coding: utf-8

import json
import urllib2

def do_request(req_url):
	try:
		f = open(req_url.replace('/','-'), 'r')
		text = f.read()
		f.close()
		return text
	except Exception as i:
		print i
		try:
			r = urllib2.urlopen(req_url).read()
			f = open(req_url.replace('/','-'),'w')
			f.write(r)
			f.close()
			return r
		except Exception as e:
			print e
			return ""


semiURL = "https://api.github.com/users/"

userName = raw_input("Digite o nome de usuário: ")

resposta = do_request(semiURL+userName)
dados = json.loads(resposta)

result = "Usuario da consulta: " + str(dados["name"]) + "\n\n"
result += str(dados["followers"]) + " seguidores:\n\n"

resposta = do_request(dados["followers_url"])
dados = json.loads(resposta)

for d in dados:
	resp = do_request(d["url"])
	datas = json.loads(resp)
	
	if(datas["name"] != None):
		result += datas["name"]
		result += "\n"
	
	resp = do_request(d["repos_url"])
	datas = json.loads(resp)
	
	for dt in datas:
		result += "\t" + str(dt["name"]) + "\n"
		
print result
